_posts/2021/2021-10-04-use-sidekiq-directly-not-through-active-job.md

```diff
@@ -0,0 +1,79 @@
+---
+title: "Use Sidekiq Directly, Not Through Active Job"
+description: "Active Job is great, but you might be missing out."
+layout: article
+category: ruby
+image:
+  base: '2021/use-sidekiq-directly-not-through-active-job'
+  alt: "Silhouetted karate kick on the beach"
+  credit: "Jason Briscoe"
+  source: "https://unsplash.com/photos/HN_4K2diUWs"
+
+---
+
+If you build a web application you should minimize the amount of time spent responding to every user; a fast website means a happy user. One way of doing that is to run long-running, parallelisable, or potentially slow, work asynchronously outside of the immediate web request. This might be sending emails, scheduled cleanups, long running calculations or anything that uses an external API.
+
+[Active Job](https://guides.rubyonrails.org/active_job_basics.html) is the recommended way of enqueuing and running background work in Rails. It provides a wrapper around a bunch of [different implementations](https://api.rubyonrails.org/classes/ActiveJob/QueueAdapters.html) of the concept. Each implementation utilises a different underlying technology and has it's own pros and cons. Most of the implementations existed before [Active Job was released](https://guides.rubyonrails.org/4_2_release_notes.html#active-job) in Rails 4.2.
+
+One of the most popular, battle-tested and performant asynchronous frameworks is [Sidekiq](https://sidekiq.org), which is multi-threaded and utilises Redis for it's queuing storage. It’s an open source library with two paid tiers for additional high-end features.
+
+Disclaimer: I'm a massive fan and “Pro” customer of Sidekiq and have met [Mike](https://twitter.com/getajobmike) (it's author) a little at a couple of Ruby conferences. That hasn't coloured my view of the library, it has _always_ been terrific and is worth every penny for our use at [work](https://coveragebook.com).
+
+
+## Instead of…
+
+…defining your jobs using Active Job:
+
+```ruby
+class DoThingsInBackgroundJob < ApplicationJob
+  queue_as :default
+
+  def peform(an_active_record_object)
+    an_active_record_object.do_things
+  end
+end
+```
+
+## Use…
+
+…Sidekiq directly:
+
+```ruby
+class DoThingsInBackgroundJob
+  include Sidekiq::Worker
+  sidekiq_options queue: "default"
+
+  def peform(id)
+    an_active_record_object = ActiveRecordObject.find_by(id: id)
+    an_active_record_object.do_things
+  end
+end
+```
+
+
+## Why?
+
+Active Job is a solid super-set of Sidekiq's functionality, but you’re missing out on a number of advantages if you use it without.
+
+If you're doing _a lot_ of work in the background, enqueuing many small fast-running jobs (as is generally suggested) there is a [2—20x performance improvement](https://github.com/mperham/sidekiq/wiki/Active-Job#performance) if you use Sidekiq directly. This improvement will be dependant on your specific setup.
```

Comment:
*comma needed after closing parenthesis 

*The level of improvement will be dependant...